.github/workflows/_snap-release.yaml

```diff
@@ -0,0 +1,63 @@
+on:
+  workflow_call:
+    inputs:
+      channel:
+        description: |
+          The channel to which charm should be released. If specified (non-empty string),
+          the snap will be released to the input channel. If kept empty, the channel will
+          be auto-selected base on the current branch. E.g. if the current branch is
+          master or main, the snap will be released to latest/edge; otherwise, the snap will
+          be released to <branch-name>/edge.
+        default: ''
```

Comment:
I think that `edge` should be the default value and we should get track from branch name, so instead having multiple steps we can have one which:
- get track from branch name if it's `main` or `master` it will be `latest`, or even better `` which will push to main track of snap
- pushed snap to `${{ track }}`/`${{ inputs.channel }}`

What do you think?